src/main/java/com/griddynamics/cd/nrp/internal/uploading/impl/ArtifactUpdateApiClientImpl.java

```diff
@@ -51,13 +48,23 @@
     private ConfigurationsManager configurationsManager;
 
     /**
+     * Queue for async requests
+     */
+    private BlockingQueue<Runnable> queue;
+
+    /**
      * ExecutorService shares between clients. All treads are created in the same executor
      */
-    private ExecutorService asyncRequestsExecutorService = Executors.newFixedThreadPool(10);
+    private ExecutorService asyncRequestsExecutorService;
 
     @Inject
     public ArtifactUpdateApiClientImpl(ConfigurationsManager configurationsManager) {
         this.configurationsManager = configurationsManager;
+        this.queue = new LinkedBlockingQueue<>(configurationsManager.getConfiguration().getRequestsQueueSize());
+        this.asyncRequestsExecutorService = new ThreadPoolExecutor(
+                configurationsManager.getConfiguration().getRequestsSendingThreadsCount(),
+                configurationsManager.getConfiguration().getRequestsSendingThreadsCount(),
+                30, TimeUnit.SECONDS, queue);
```

Comment:
Well, you know the problem with the unbounded queues - they are unbounded :) Stability of Nexus should be considered as a priority while unbounded queues is a risk to get OOM.